OCaml 5.1 support

There are a few errors in balmap:

```
File "balmap/map.ml", lines 12-371, characters 0-3:
 12 | struct
 13 |   type key = O.t
 14 | 
 15 |   type 'a t = (O.t, 'a) balmap
 16 | 
...
368 | 
369 |   let of_seq seq =
370 |     add_seq seq empty
371 | end
Error: Signature mismatch:
       ...
       The value `add_to_list' is required but not provided
       File "map.mli", line 88, characters 4-56: Expected declaration
       The value `to_list' is required but not provided
       File "map.mli", line 333, characters 4-41: Expected declaration
       The value `of_list' is required but not provided
       File "map.mli", line 337, characters 4-41: Expected declaration
```

and physh:

```
In file included from alloc.h:21,
                 from ml_physh_map.c:6:
ml_physh_map.c: In function ‘ml_physh_map_alloc’:
ml_physh_map.c:64:25: error: ‘caml_stat_minor_collections’ undeclared (first use in this function); did you mean ‘caml_stat_major_collections’?
   64 |   Field(v, 0) = Val_int(caml_stat_minor_collections);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/balmap/map.ml
+++ b/balmap/map.ml
@@ -368,4 +368,11 @@ struct
 
   let of_seq seq =
     add_seq seq empty
+
+  let add_to_list x data m =
+    let add = function None -> Some [data] | Some l -> Some (data :: l) in
+    update x add m
+
+  let to_list = bindings
+  let of_list bs = List.fold_left (fun m (k, v) -> add k v m) empty bs
 end

--- a/balmap/set.ml
+++ b/balmap/set.ml
@@ -366,4 +366,6 @@ struct
         Bt1.node l k1 r
       else
         Bt1.join l r
+
+  let to_list = elements
 end

